subsys/net/ip/net_if.c

```diff
@@ -236,11 +236,22 @@ static inline void init_iface(struct net_if *iface)
 {
 	const struct net_if_api *api = net_if_get_device(iface)->driver_api;
 
-	NET_ASSERT(api && api->init && api->send);
+	NET_ASSERT(api && api->init);
 
 	NET_DBG("On iface %p", iface);
 
 	api->init(iface);
+
+	/* test for api->send only when ip is *not* offloaded: */
+#if defined(CONFIG_NET_OFFLOAD)
```

Comment:
Well, I guess this is a perfect case to use IS_ENABLED() to avoid repetition. Or maybe we just should define net_if_is_ip_offloaded() to return the right thing for !CONFIG_NET_OFFLOAD case and have even clearer flow.

Reply:
Yes, @jukkar, actually that would help implement @pfalcon's second suggestion, simplifying to:

`if (!net_if_is_ip_offloaded(iface)) {
		NET_ASSERT(api->send);
}`

Reason I didn't do that initially was because it seemed net_core favored code size reduction via Kconfig vs readability.   But, I much prefer the readability of the run-time check.

How close is #9825 from  being merged?  Would you prefer I grab commit 1ca71cf as part of this PR, or wait and rebase?

Reply:
I just sent this as a separate PR, see #10521

Reply:
Perfect, thanks!